File: app/src/main/java/com/codelabs/accountsafetyfido/java/util/fingerprint/FingerprintHelper.java
/*
 *  Copyright 2020. Huawei Technologies Co., Ltd. All rights reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package com.codelabs.accountsafetyfido.java.util.fingerprint;

import android.content.Context;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

/**
 * Created by CodeLab
 */
public final class FingerprintHelper {

    /**
     * save String data
     *
     * @param key   save key
     * @param value save value
     */
    public static void put(Context context, String key, String value) {
        File file = new File(context.getFilesDir(), key.hashCode() + "");
        BufferedWriter out = null;
        try {
            out = new BufferedWriter(new FileWriter(file), 1024);
            out.write(value);
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            if (out != null) {
                try {
                    out.flush();
                    out.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }


    /**
     * get data
     *
     * @param key save key
     * @return String value
     */
    public static String get(Context context, String key) {
        File file = new File(context.getFilesDir(), key.hashCode() + "");
        if (!file.exists()) {
            return null;
        }
        BufferedReader in = null;
        try {
            in = new BufferedReader(new FileReader(file));
            StringBuilder readString = new StringBuilder();
            String currentLine;
            while ((currentLine = in.readLine()) != null) {
                readString.append(currentLine);
            }
            return readString.toString();
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        } finally {
            if (in != null) {
                try {
                    in.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }

    /**
     * delete data
     *
     * @param key save key
     */
    public static void delete(Context context, String key) {
        File file = new File(context.getFilesDir(), key.hashCode() + "");
        if (file.exists()) {
            file.delete();
        }
    }
}

File: app/src/main/java/com/codelabs/accountsafetyfido/kotlin/util/fingerprint/FingerprintHelper.kt
/*
 *  Copyright 2020. Huawei Technologies Co., Ltd. All rights reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package com.codelabs.accountsafetyfido.kotlin.util.fingerprint

import android.content.Context
import java.io.*

/**
 * Created by CodeLab
 */
object FingerprintHelper {

    /**
     * save String data
     *
     * @param key   save key
     * @param value save value
     */
    @JvmStatic
    fun put(context: Context, key: String, value: String?) {
        val file = File(context.filesDir, key.hashCode().toString() + "")
        var out: BufferedWriter? = null
        try {
            out = BufferedWriter(FileWriter(file), 1024)
            out.write(value)
        } catch (e: IOException) {
            e.printStackTrace()
        } finally {
            try {
                out!!.flush()
                out.close()
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }
    }

    /**
     * get data
     *
     * @param key save key
     * @return String value
     */
    @JvmStatic
    operator fun get(context: Context, key: String): String? {
        val file = File(context.filesDir, key.hashCode().toString() + "")
        if (!file.exists()) {
            return null
        }
        var bufferedReader: BufferedReader? = null
        return try {
            bufferedReader = BufferedReader(FileReader(file))
            val readString = StringBuilder()
            var currentLine: String?
            while (bufferedReader.readLine().also { currentLine = it } != null) {
                readString.append(currentLine)
            }
            readString.toString()
        } catch (e: IOException) {
            e.printStackTrace()
            null
        } finally {
            try {
                bufferedReader!!.close()
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }
    }

    /**
     * delete data
     *
     * @param key save key
     */
    @JvmStatic
    fun delete(context: Context, key: String) {
        val file = File(context.filesDir, key.hashCode().toString() + "")
        if (file.exists()) {
            file.delete()
        }
    }
}